snmp/datadog_checks/snmp/snmp.py

```diff
@@ -329,13 +332,36 @@ def _consume_binds_iterator(self, binds_iterator, config):
             all_binds.extend(var_binds_table)
         return all_binds, error
 
+    def _start_discovery(self):
+        cache = read_persistent_cache(self.check_id)
+        if cache:
```

Comment:
Do we need a mechanism to flush old cached hosts in `discovered_instances` ? In case a host is removed from the network.

Seems this invalidation only check for `host` format:

```
                try:
                    ipaddress.ip_address(host)
                except ValueError:
                    write_persistent_cache(self.check_id, json.dumps([]))
                    break
```

Reply:
Yes the mechanism is here: https://github.com/DataDog/integrations-core/blob/c37575a06f8b3798930355805eeba880221ac2f5/snmp/datadog_checks/snmp/snmp.py#L344. Then we persist it during the discovery process as usual.